[Draft] Various fixes

TheBv opened the pull request:
This is a draft because I'll continue adding more fixes and the likes over the next couple of days.

What's new:
I added a path finder for creatures that can climb (e.g myrmex) since they, with the current implementation, can't.
Since multiple creatures and not just dragons use that navigator I changed the name of the class DragonAdvancedPathNavigate to AdvancedPathNavigate since it's not exclusively being used by dragons. 
I also made it so you can parse a MovementType to that class eg. CLIMBING, FLYING.
This is a suggestive implementation and I can approach this differently if people want me to.
 
Current fixes:

- Fixed further casting issues related to myrmex which could appear when they are riding another entity. #3275
- Implemented further fixes to reduce TPS issues related to myrmex when not finding any items/blocks or not being able to store anything. This was achieved by making them wait a random amount of ticks from 0-20 when they couldn't successfully do one of the actions. This might need some fine tuning. Lower numbers make them more responsive but cause a bit more lag while higher ones make them more unresponsive but the lag gets reduced (lag <-> tps)
- Fixed dread-torches not having a texture when placed on a wall
- Hippogryphs can now breed again #3385
- Should have fixed most issues with biome files #3381
- Fixed #3347/ all issues related to entities that have a chest-slot
- isFarEnoughFromDangerousGen doesn't use deprecated methods anymore
- Fixed icedragons not considering isFarEnoughFromDangerousGen
- Fixed icedragons being able to spawn older then it was intended
- Fixed pixie jar issues that were related to nbt data #3343
- Implemented toLowerCase(Locale.ROOT) where it was needed #3345
- Fixed an entity, using the advancedpathnavigator, causing issues when having a passenger  #3393,#3394

Things I'm looking into/ are important:

-

Notes: As far as I'm aware changing the damage values on items/ anything that has to be registered is "impossible" or requires some ugly workarounds so I won't fix that. (#3389)

I'll keep updating this list as time goes on.

TheBv committed:
First steps to generialzing Advanced Path Navigate

TheBv committed:
Myrmex can again climb blocks like spiders.
Further implementation of the more generialzed AdvancedPathNavigator

TheBv committed:
Merge pull request #7 from TheBv/1.16.3

1.16.3

TheBv committed:
Fixed myrmex casting issues which were caused by an entity having a myrmex as a rider.
Further implemented ways to reduce TPS issues in case myrmex can't find anything to forage.

TheBv committed:
Fixed dread torch and burnt torch having a missing texture on the wall.
In previous commit: fixed further myrmex casting issues caused by them riding something and wanting to execute their (Re)EnterHive tasks.
First implementation of a climbing-path navigator which let's entities that have the AdvancedPathNaviagtor pathfind with having in mind that they can climb blocks.

TheBv committed:
Hippogryph can now breed again properly

TheBv committed:
Merge pull request #8 from TheBv/pathrewrite

Pathrewrite

TheBv committed:
Removed useless code

TheBv committed:
Removed unnecessary classes.
Hopefully fixed #3381 by making sure that if only biome names are being used eg ["minecraft:desert_hills"] it only checks anyBiomeInfoInList().
Otherwise exclude,mustInclude,include where all empty (since non of the operators where used)
and therefor it'd return true.

TheBv committed:
Fixed #3347

TheBv committed:
Changed isFarEnoughFromDangerousGen to not use a deprecated method.

TheBv committed:
Fixed ice dragon roots not implementing isFarEnoughFromDangerousGen
Fixed ice dragon roots spawning a dragon of unintended age.

TheBv committed:
Small revert

TheBv committed:
Fixed #3343

TheBv committed:
Implemented #3345 where necessary

TheBv committed:
Fixed small issue with the advancedpathnavigator climbing.
Removed maximumWanderRadius in favor of a waiting system.
Waiting system is now also being used in MyrmexAIStoreItems, if no cocoon/ no empty cocoon was found.

TheBv committed:
debug false

TheBv committed:
Fixed #3394,#3393 and similar issues

TheBv committed:
MyrmexAI now also checks if the navigator is an instance of AdvancedPathNavigate just to be sure

TheBv commented:
I think that's it. As far as I'm aware I've looked at every major issue I could find and fixed it.
I guess if something pops up in the next couple of days I'll attend to it but otherwise I'm happy with this.
Thanks to all the people that report these issues btw!

Daejangnamu commented:
[crash-2021-01-08_13.34.14-server.txt](https://github.com/Alex-the-666/Ice_and_Fire/files/5786747/crash-2021-01-08_13.34.14-server.txt)

server crash

TheBv committed:
Fixed myrmex being able to attack players with a reputation above 25 properly now.

TheBv committed:
Hopefully fixed crash report related to minecraft version 1.16.4
Fixed issues with climbing path-navigator

TheBv committed:
Tweaked some numbers so myrmex cause less tps issues
MyrmexAIAttackMelee: shouldContinueExecuting now considers lastAttackedEntity which should make the behavior of myrmex attacking a lot less strange.

Daejangnamu commented:
Works fine. Ty

TheBv committed:
Fixed Hippcampus/Hippogryph issues when being saddled etc
